src/cosalib/aws.py

```diff
@@ -191,4 +193,6 @@ def aws_cli(parser):
     parser.add_argument("--grant-user-snapshot", help="Grant user snapshot volume permission",
                         nargs="*", default=[])
     parser.add_argument("--public", action="store_true", help="Mark images as publicly available")
+    parser.add_argument("--tags", help="list of key=value tags to attach to the AMI",
+                        nargs="?", default=[], action='append')
```

Comment:
I don't think we should use `nargs="?"` here. It should work as expected if you just drop that.

Reply:
And while I look at this, I really think we should replace the other `nargs="*"` in this parser with `action='append'`. That's technically a breaking change, but the pipeline is already passing it the right way so shouldn't be broken by that: https://github.com/coreos/fedora-coreos-pipeline/blob/a2e854aed78b7eb68808199f8845a78455bcd666/libcloud.groovy#L149-L152

Anyway, should probably do that in a follow-up instead.

Reply:
Totally Agree. I was telling @gursewak1997 to match what was being done for grant-user, but apparently that was wrong (but only worked because `nargs=` forced a list to be returned)! 

I agree `action='append'` should be used here in both cases.

I'd argue let's fix it now and not in a followup so we can test them both.